src/app/components/AccountHistory.tsx

```diff
@@ -198,6 +204,32 @@ function DisplayTable() {
   );
   const openOrders = useAppSelector(selectOpenOrders);
   const orderHistory = useAppSelector(selectOrderHistory);
+  const dispatch = useAppDispatch();
+  const hideOtherPairs = useAppSelector(
+    (state) => state.accountHistory.hideOtherPairs
+  );
+  const combinedOrderHistory = useAppSelector(selectCombinedOrderHistory);
+
+  useEffect(() => {
+    const showAllPairs = !hideOtherPairs;
+    if (showAllPairs) {
+      dispatch(fetchAccountHistoryAllPairs());
+    }
+
+    const intervalId = setInterval(() => {
+      if (showAllPairs) {
+        dispatch(fetchAccountHistoryAllPairs());
+      }
+    }, 120000); // Dispatch every 2 mins (120 seconds)
+
+    return () => clearInterval(intervalId); // Cleanup interval on component unmount
+  }, [dispatch, hideOtherPairs]);
```

Comment:
Can we move this code one layer up to be right after [this code](https://github.com/DeXter-on-Radix/website/blob/main/src/app/trade/page.tsx#L45-L49)?